app/components/goodcity/add-item-overlay.js

```diff
@@ -0,0 +1,99 @@
+import Ember from "ember";
+const { getOwner } = Ember;
+import SingletonComponent from "../base/global";
+import _ from "lodash";
+
+export default SingletonComponent.extend({
```

Comment:
No need to inherit from `SingletonComponent`, that was made only for the global ones. You can inherit from `Ember.Component` since they are used within the pages and therefore get destroyed alongside it

Reply:
Applies to other components too : )

Reply:
Other note: It cannot be **both** bound to a service **and** within a page's`.hbs` file.

That will break, since the page gets destroyed but the service remains.

Options:

* 1) Component is inside a page's template, and is bound to the controller _(normal kind)_  
* 2) Component is inside the `_global_components.hbs` template and is bound to a service